scripts/bazel_diff.py

```diff
@@ -0,0 +1,170 @@
+        f"--script_path={bazel_diff_path}",
+        "//bazel/diff:bazel-diff",
+    ]
+    p = subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8"
+    )
+    if p.returncode != 0:
+        print(p.stderr)
+        exit(f"Bazel run returned {p.returncode}")
+    return bazel_diff_path
+
+
+def compute_hashes(
+    bazel: str, bazel_diff: str, tmpdir: str, prefix: str
+) -> str:
+    hashes_path = os.path.join(tmpdir, f"{prefix}_hashes")
+    args = [
+        bazel_diff,
+        "generate-hashes",
+        f"-b={bazel}",
+        f"-w={os.getcwd()}",
+        hashes_path,
+    ]
+    p = subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8"
+    )
+    if p.returncode != 0:
+        print(p.stderr)
+        exit(f"Bazel diff returned {p.returncode}")
+    return hashes_path
+
+
+def impacted_targets(
+    bazel_diff: str, baseline_hashes: str, current_hashes: str
+) -> str:
+    args = [
+        bazel_diff,
+        "get-impacted-targets",
+        f"-sh={baseline_hashes}",
+        f"-fh={current_hashes}",
+    ]
+    p = subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8"
+    )
+    if p.returncode != 0:
+        print(p.stderr)
+        exit(f"Bazel diff returned {p.returncode}")
+    return p.stdout
+
+
+def filter_targets(bazel: str, targets: str) -> str:
+    args = [
+        bazel,
+        "query",
+        (
+            f"let t = set({targets}) in "
+            f"kind(rule, $t) except attr(tags, manual, $t)"
+        ),
+    ]
+    p = subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8"
+    )
+    if p.returncode != 0:
+        print(p.stderr)
+        exit(f"Bazel run returned {p.returncode}")
+    return p.stdout
+
+
+def git_checkout(commit: str) -> None:
+    subprocess.run(
+        [
+            "git",
+            "checkout",
+            "--quiet",
+            commit,
+        ],
+        check=True,
+        encoding="utf-8",
+    )
+
+
+def git_diff(baseline: str, current: str) -> None:
+    subprocess.run(
+        [
+            "git",
+            "diff",
+            "--stat",
+            f"{baseline}..{current}",
+        ],
+        check=True,
+        encoding="utf-8",
+    )
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(__doc__)
+    parser.add_argument(
+        "--baseline", required=True, help="Git commit of the diff baseline."
+    )
+    parser.add_argument(
+        "--current", required=True, help="Git commit of the current state."
+    )
+    parsed_args = parser.parse_args()
+
+    scripts_utils.chdir_repo_root()
+    bazel = scripts_utils.locate_bazel()
+
+    baseline = parsed_args.baseline
+    current = parsed_args.current
+    log(f"Diffing Bazel targets for {baseline}..{current}")
+    git_diff(baseline, current)
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        tmpdir = tempfile.mkdtemp()
```

Comment:
This call is redundant, you are already creating a temporary directory with the above call to TemporaryDirectory